tests/unit_tests/test_backend_utils.py

```diff
@@ -0,0 +1,123 @@
+import pathlib
+from typing import Dict
+from unittest.mock import Mock
+from unittest.mock import patch
+
+import pytest
+
+from sky import clouds
+from sky import skypilot_config
+from sky.backends import backend_utils
+from sky.resources import Resources
+from sky.resources import resources_utils
+
+
+@patch.object(skypilot_config, 'CONFIG_PATH',
+              './tests/test_yamls/test_aws_config.yaml')
+@patch.object(skypilot_config, '_dict', None)
+@patch.object(skypilot_config, '_loaded_config_path', None)
+@patch('sky.clouds.service_catalog.instance_type_exists', return_value=True)
+@patch('sky.clouds.service_catalog.get_accelerators_from_instance_type',
+       return_value={'fake-acc': 2})
+@patch('sky.clouds.service_catalog.get_image_id_from_tag',
+       return_value='fake-image')
+@patch.object(clouds.aws, 'DEFAULT_SECURITY_GROUP_NAME', 'fake-default-sg')
+@patch('sky.check.get_cloud_credential_file_mounts',
+       return_value='~/.aws/credentials')
+@patch('sky.backends.backend_utils._get_yaml_path_from_cluster_name',
+       return_value='/tmp/fake/path')
+@patch('sky.utils.common_utils.fill_template')
+def test_write_cluster_config_w_remote_identity(mock_fill_template,
```

Comment:
new test for remote identity, this is somewhat redundant with the resource_utils test since it is encapsulated in `write_cluster_config`.

I can remove that test if we desire.